### ISSUE TITLE
Move buildings layer to fix associated issues

### ISSUE THREAD

**Issue by jragusa** [gravitystorm/openstreetmap-carto#3583]
The rendering of underground building #552 raises the question of the relative position of the `buildings` layer which is also discussed in several others issues:
* #688 with regard highways
* #2037 and #2054 with regard to `railway=platform`
* #2475 about `building=roof`
* #1285 with regard to military overlay and prison pattern (see [comment](https://github.com/gravitystorm/openstreetmap-carto/issues/1285#issuecomment-447043645))

Resolving one after one each of these problems will lead to continuous corrections of the layers with respect to the buildings one. So, we need to discuss what would be the correct position of the `buildings` layer to fix in one shot all of these problems. 

Basically, we need to move `stations-poly` and all highway related layers before (e.g. below) `buildings`  layer but what about underground building in this case ? There is also some aside problems such as the lack of rendering of roof parking (only icon is rendered).

**Comment by jeisenbe** [comment]
Closing this meta-issue; see linked issues for more specific discussion: #688 (highway areas)
#1285 (military and prison overlays)
#2037 and #2054 (railway=platform), #2475 (building=roof)
